Crash report "java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $"

UUID: 69307c29eda98d631ec6282a589b0e5a
Build flavor: stable
Version name: v4.11.1
OS info: Android 7.0, sdk version: 24
Report description: No title
Logs:
```
io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.ConsumerSingleObserver.onError(ConsumerSingleObserver.java:46)
	at io.reactivex.internal.operators.single.SingleObserveOn$ObserveOnSingleObserver.run(SingleObserveOn.java:79)
	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:124)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6688)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226)
	at com.google.gson.Gson.fromJson(Gson.java:932)
	at com.google.gson.Gson.fromJson(Gson.java:897)
	at com.google.gson.Gson.fromJson(Gson.java:846)
	at com.google.gson.Gson.fromJson(Gson.java:817)
	at com.github.adamantcheese.chan.core.repository.SavedThreadLoaderRepository.loadOldThreadFromJsonFile(SavedThreadLoaderRepository.kt:49)
	at com.github.adamantcheese.chan.core.manager.SavedThreadLoaderManager.loadSavedThread(SavedThreadLoaderManager.kt:72)
	at com.github.adamantcheese.chan.core.site.loader.ChanThreadLoader.loadSavedThreadIfItExists(ChanThreadLoader.java:617)
	at com.github.adamantcheese.chan.core.site.loader.ChanThreadLoader.lambda$onErrorResponse$12$ChanThreadLoader(ChanThreadLoader.java:553)
	at com.github.adamantcheese.chan.core.site.loader.-$$Lambda$ChanThreadLoader$v8zPBBoLGz-KfLh0MEsTzsnNe-8.call(lambda)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3603)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:761)
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:386)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215)
	... 20 more
```

--- a/Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/loader/ChanThreadLoader.java
+++ b/Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/loader/ChanThreadLoader.java
@@ -45,6 +45,7 @@ import com.github.adamantcheese.chan.core.site.parser.ChanReaderRequest;
 import com.github.adamantcheese.chan.ui.helper.PostHelper;
 import com.github.adamantcheese.chan.utils.BackgroundUtils;
 import com.github.adamantcheese.chan.utils.Logger;
+import com.google.gson.JsonSyntaxException;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -359,7 +360,7 @@ public class ChanThreadLoader
 
         Disposable disposable = Single.fromCallable(() -> onResponseInternal(response))
                 .subscribeOn(Schedulers.io())
-                .subscribe(result -> {}, error -> Logger.e(TAG, "onResponse error", error));
+                .subscribe(result -> { }, error -> Logger.e(TAG, "onResponse error", error));
 
         compositeDisposable.add(disposable);
     }
@@ -577,26 +578,47 @@ public class ChanThreadLoader
                             chanThread.isClosed(),
                             chanThread.isArchived()
                     );
+
+                    // We managed to load local thread, do no need to show the error screen
                     return false;
                 }
             }
 
+            // No local thread, show the error screen
             return true;
-        }).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(result -> {
-            if (!result) {
-                return;
-            }
+        }).subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(showError -> {
+                    if (!showError) {
+                        return;
+                    }
 
-            Logger.i(TAG, "Loading error", error);
-            clearTimer();
-            ChanLoaderException loaderException = new ChanLoaderException(error);
+                    Logger.i(TAG, "Loading error", error);
+                    notifyAboutError(error);
+                }, throwable -> {
+                    Logger.i(TAG, "Loading unhandled error", throwable);
 
-            for (ChanLoaderCallback l : listeners) {
-                l.onChanLoaderError(loaderException);
-            }
-        });
+                    notifyAboutError(createError(throwable));
+                });
 
         compositeDisposable.add(disposable);
+    }
+
+    private VolleyError createError(Throwable throwable) {
+        if (throwable instanceof JsonSyntaxException) {
+            return new VolleyError("Error while trying to load local thread", throwable);
+        }
+
+        return new VolleyError("Unhandled exception", throwable);
+    }
+
+    private void notifyAboutError(VolleyError error) {
+        clearTimer();
+        ChanLoaderException loaderException = new ChanLoaderException(error);
+
+        for (ChanLoaderCallback l : listeners) {
+            l.onChanLoaderError(loaderException);
+        }
     }
 
     /**

